Translate passing data deeply with context

dekimiq opened the pull request.

dekimiq committed:
translated from start to title step1

dekimiq committed:
translated from step1 to title context passes...

dekimiq committed:
translated from 'context passes...' to end

dekimiq committed:
review and correction

dekimiq committed:
fixing a bug with the start navigation menu (header is too big, when switching to the header, it does not show up active in the menu)

titovmx reviewed (changes requested):
Спасибо за перевод! Оставил несколько предложений

dekimiq committed:
Commit suggestion - 7 line

Commit suggestion - 7 line (from "компонентам" to "компонентов")

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 22 line

Commit suggestion - 22 line (from "через UI дерево компонентам, которые их используют" to "по дереву компонентов туда, где они используются")

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 24 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 41 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 315 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 841 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Commit suggestion - 850 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
Merge branch 'reactjs:main' into translate-passing-data-deeply-with-context

dekimiq committed:
Commit suggestion - 35 line

Co-authored-by: Maxim Titov <mtitov92@gmail.com>

dekimiq committed:
corrections have been made about the drilling of props

dekimiq committed:
translate attribute - alt on diagrams

dekimiq committed:
check and correcting symbol 'ё'

Pyplee commented on src/content/learn/passing-data-deeply-with-context.md:
```diff
@@ -1,48 +1,48 @@
-- How to replace repetitive prop passing with context
-- Common use cases for context
-- Common alternatives to context
+- Что такое "проп бурение"
```
Я посмотрел, и правда, зачастую разработчики пишут prop drilling, иногда проставляя рядом название явления на Русском. Я исправил этот недочет. 

Я поставил ("prop drilling" (бурение пропсов)) в раздел - вы узнаете\выучите (13 строка), принял исправление на 35 (написал просто "Бурение пропсов") и поставил полноценное расшифрование ("prop drilling" (бурение пропсов)) на 880 строке т.к она является заголовком в челленджах, думаю не будет лишним.

Pyplee commented on src/content/learn/passing-data-deeply-with-context.md:
```diff
@@ -1,48 +1,48 @@
 
 <DiagramGroup>
 
 <Diagram name="passing_data_lifting_state" height={160} width={608} captionPosition="top" alt="Diagram with a tree of three components. The parent contains a bubble representing a value highlighted in purple. The value flows down to each of the two children, both highlighted in purple." >
```
Сделал

Pyplee commented:
Заметки внес, поправил некоторые моменты, прошу проревьюрить перевод.

dimatitov commented:
👋 @Pyplee, спасибо за проделанную работу!

Вижу, что в PR сейчас есть конфликты — потребуется их разрешить, чтобы можно было продолжить.

@titovmx, когда будет возможность, посмотри, пожалуйста, актуальное состояние — возможно, после разрешения конфликтов PR уже готов к мержу.
